src/libjvm.rs

```diff
@@ -0,0 +1,74 @@
+use crate::GlobalResult;
+
+/// Virtual table for top-level libjvm functions that create or get JVMs.
+#[allow(non_snake_case)]
+pub(crate) struct Libjvm {
+    pub JNI_CreateJavaVM: unsafe extern "system" fn(
+        pvm: *mut *mut jni_sys::JavaVM,
+        penv: *mut *mut std::ffi::c_void,
+        args: *mut std::ffi::c_void,
+    ) -> jni_sys::jint,
+    pub JNI_GetCreatedJavaVMs: unsafe extern "system" fn(
+        vmBuf: *mut *mut jni_sys::JavaVM,
+        bufLen: jni_sys::jsize,
+        nVMs: *mut jni_sys::jsize,
+    ) -> jni_sys::jint,
+}
+
+#[cfg(feature = "dylibjvm")]
+mod dynlib {
+    use std::path::{Path, PathBuf};
+
+    use libloading::Library;
+    use once_cell::sync::OnceCell;
+
+    use super::*;
+    use crate::Error;
+
+    static LIBJVM: OnceCell<Libjvm> = OnceCell::new();
+
+    #[allow(non_snake_case)]
+    fn load_libjvm_at(path: &Path) -> GlobalResult<Libjvm> {
+        (|| {
+            let lib = unsafe { Library::new(path) }?;
+            let JNI_CreateJavaVM = *unsafe { lib.get(b"JNI_CreateJavaVM\0") }?;
+            let JNI_GetCreatedJavaVMs = *unsafe { lib.get(b"JNI_GetCreatedJavaVMs\0") }?;
+            Ok(Libjvm {
+                JNI_CreateJavaVM,
+                JNI_GetCreatedJavaVMs,
+            })
```

Comment:
I investigated a bit the `SIGSEGV` in CI: it's because `Library` [unloads the library on drop](https://docs.rs/libloading/latest/libloading/struct.Library.html#method.close).